preload.php

```diff
@@ -2,257 +2,262 @@
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/StaticVar.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/Static_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/Switch_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/Throw_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/TraitUse.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/TraitUseAdaptation/Alias.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/TraitUseAdaptation/Precedence.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/Trait_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/TryCatch.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/Unset_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/UseUse.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/Use_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/Stmt/While_.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/UnionType.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/VarLikeIdentifier.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Node/VariadicPlaceholder.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeDumper.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeFinder.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeVisitor/CloningVisitor.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeVisitor/FindingVisitor.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeVisitor/FirstFindingVisitor.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeVisitor/NameResolver.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeVisitor/NodeConnectingVisitor.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/NodeVisitor/ParentConnectingVisitor.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Parser/Multiple.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Parser/Php5.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Parser/Php7.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/Parser/Tokens.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/ParserFactory.php';
-require_once __DIR__ . '/vendor/nikic/php-parser/lib/PhpParser/PrettyPrinter/Standard.php';
+// prevent re-declare class/interface errors, because of a already loaded php-parser package
+if (!interface_exists('\PhpParser\Node')) {
```

Comment:
`preload.php` is generated by `composer preload`, which should not be manually touched, the command is at:

https://github.com/rectorphp/rector-src/blob/685ba8a63e904c776033ba85c13f8903952e73e4/build/build-preload.php#L44

Reply:
ohh my .. another problem is, that because of rector custome patches to nikic/php-parser rector cannot work with a external php-parser version, because the external php-parser will likely not contain the patches rector needs/wants.

IMO rector would need to declare a conflict with any nikic/php-parser as it effectively will not work with another versions loaded at the same time

Reply:
yes, I think `conflict` config can be solution 👍

Reply:
or if the issue only in phpunit, then, probably the patch should be in phpunit part itself?

Reply:
`conflict` seems possibly the solution, as there are new nodes that renamed on v5

https://github.com/rectorphp/rector-php-parser/blob/9639ef9429f74c14fdc3c7dad9d3b0d916663de0/config/sets/php-parser-50.php#L18